server/issue_test.go

```diff
@@ -279,12 +269,214 @@ func TestRouteShareIssuePublicly(t *testing.T) {
+		expectedErr error
+	}{
+		"Success": {
+			instanceID: testInstance1.InstanceID,
+			client:     testClient{},
+			connection: &Connection{
+				Settings: &ConnectionSettings{
+					Notifications: true,
+					RolesForDMNotification: map[string]bool{
+						assigneeRole: true,
+						mentionRole:  true,
+						reporterRole: true,
+						watchingRole: true,
+					},
+				},
+				User: jira.User{
+					AccountID: "test-AccountID",
+				},
+			},
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{
+						Fields: &jira.IssueFields{
+							Creator: &jira.User{},
+						},
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		"Issue Field not found": {
+			instanceID: testInstance1.InstanceID,
+			client:     nil,
+			connection: nil,
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{},
+				},
+			},
+			expectedErr: nil,
+		},
+		"Unable to load instance": {
+			instanceID: "test-instanceID",
+			client:     nil,
+			connection: nil,
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{
+						Fields: &jira.IssueFields{
+							Creator: &jira.User{},
+						},
+					},
+				},
+			},
+			expectedErr: errors.New(fmt.Sprintf("instance %q not found", "test-instanceID")),
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			client, connection, error := tt.wh.fetchConnectedUser(p, tt.instanceID)
+			assert.Equal(t, tt.connection, connection)
+			assert.Equal(t, tt.client, client)
+			if tt.expectedErr != nil {
+				assert.Error(t, tt.expectedErr, error)
+			}
+		})
+	}
+}
 
-	api.On("LogError", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
+func TestApplyReporterNotification(t *testing.T) {
+	p := setupTestPlugin(&plugintest.API{})
 
-	api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
+	wh := &webhook{
+		eventTypes: map[string]bool{createdCommentEvent: true},
+		JiraWebhook: &JiraWebhook{
+			Comment: jira.Comment{
+				UpdateAuthor: jira.User{},
+			},
+			Issue: jira.Issue{
+				Key: "test-key",
+				Fields: &jira.IssueFields{
+					Type: jira.IssueType{
+						Name: "Story",
+					},
+					Summary: "",
+				},
+				Self: "test-self",
+			},
+		},
+		notifications: []webhookUserNotification{},
+	}
+	for name, tt := range map[string]struct {
+		instanceID types.ID
+		reporter   *jira.User
+	}{
+		"Success": {
+			instanceID: testInstance1.InstanceID,
+			reporter:   &jira.User{},
+		},
+		"Unable to load instance": {
+			instanceID: "test-instanceID",
+			reporter:   &jira.User{},
+		},
+		"Reporter is nil": {
+			instanceID: testInstance1.InstanceID,
+			reporter:   nil,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			totalNotifications := len(wh.notifications)
+			p.applyReporterNotification(wh, tt.instanceID, tt.reporter)
+			if tt.reporter == nil || tt.instanceID == "test-instanceID" {
+				assert.Equal(t, len(wh.notifications), totalNotifications)
+			} else {
+				assert.Equal(t, len(wh.notifications), 1+totalNotifications)
+			}
```

Comment:
Instead of having if statements that have implicit knowledge about the test data, the individual test cases should provide enough context to make it so there's no logic here. Such as having a string field `expectedError` that contains the error we expect to receive for that test case, and is empty when we expect no error.